ext/date/php_date.c

```diff
@@ -3457,6 +3464,11 @@ PHP_FUNCTION(timezone_open)
 		Z_PARAM_STR(tz)
 	ZEND_PARSE_PARAMETERS_END();
 
+	if (ZSTR_LEN(tz) != strlen(ZSTR_VAL(tz))) {
+		zend_argument_type_error(1, "cannot contain null bytes");
+		RETURN_THROWS();
+	}
+
```

Comment:
Instead of this check, we probably should use  `Z_PARAM_STR_EX(dest, 1, 0)` instead, as that checks for NULL bytes already.

Reply:
This doesn't check for null bytes within the string but if the value passed is ``null`` and not a string. Moreover there is a more explicit macro for this: ``Z_PARAM_STR_OR_NULL(dest)``